code/game/machinery/quantum_pad.dm

```diff
@@ -190,3 +196,49 @@
 /obj/item/paper/guides/quantumpad
 	name = "Quantum Pad For Dummies"
 	info = "<center><b>Dummies Guide To Quantum Pads</b></center><br><br><center>Do you hate the concept of having to use your legs, let alone <i>walk</i> to places? Well, with the Quantum Pad (tm), never again will the fear of cardio keep you from going places!<br><br><c><b>How to set up your Quantum Pad(tm)</b></center><br><br>1.Unscrew the Quantum Pad(tm) you wish to link.<br>2. Use your multi-tool to cache the buffer of the Quantum Pad(tm) you wish to link.<br>3. Apply the multi-tool to the secondary Quantum Pad(tm) you wish to link to the first Quantum Pad(tm)<br><br><center>If you followed these instructions carefully, your Quantum Pad(tm) should now be properly linked together for near-instant movement across the station! Bear in mind that this is technically a one-way teleport, so you'll need to do the same process with the secondary pad to the first one if you wish to travel between both.</center>"
+
+/obj/item/circuit_component/quantumpad
+	display_name = "Quantum Pad"
+	display_desc = "A bluespace quantum-linked telepad used for teleporting objects to other quantum pads."
+	circuit_flags = CIRCUIT_FLAG_INPUT_SIGNAL
+
+	var/datum/port/input/target_pad
+
+	var/obj/machinery/quantumpad/attached_pad
+
+/obj/item/circuit_component/quantumpad/Initialize()
+	. = ..()
+	target_pad = add_output_port("Target Pad", PORT_TYPE_ATOM)
+
+/obj/item/circuit_component/quantumpad/register_usb_parent(atom/movable/parent)
+	. = ..()
+	if(istype(parent, /obj/machinery/quantumpad))
+		attached_pad = parent
+
+/obj/item/circuit_component/quantumpad/unregister_usb_parent(atom/movable/parent)
+	attached_pad = null
+	return ..()
+
+/obj/item/circuit_component/quantumpad/input_received(datum/port/input/port)
+	. = ..()
+	if(. || !attached_pad)
```

Comment:
1. Don't copy these checks. Make a new proc in the quantum pad machine itself that returns the error message if it fails, and null if it's right. Use this in both places. You don't have to reference the error message anywhere for the circuit, that's for sending a message to the user in the current place, but you *do* have to prevent this code duplication.
2. Add an "on fail" signal, like everything else.

Reply:
You are still copying the checks. This is not resolved.